packages/protocol/contracts/governance/VestingFactory.sol

```diff
@@ -0,0 +1,72 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+import "./VestingInstance.sol";
+import "./interfaces/IVestingFactory.sol";
+
+import "../common/Initializable.sol";
+import "../common/UsingRegistry.sol";
+
+contract VestingFactory is Initializable, UsingRegistry, IVestingFactory {
+  // mapping between beneficiary addresses and associated vesting contracts (schedules)
+  mapping(address => address) public hasVestedAt;
+
+  function initialize(address registryAddress) external initializer {
+    _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
+  }
+
+  event NewVestingInstanceCreated(address atAddress);
```

Comment:
Should we include and index beneficiary?

Reply:
done